boring-sys/build.rs

```diff
@@ -155,85 +155,104 @@ const BORING_SSL_PATH: &str = "deps/boringssl";
+                    // Mac Catalyst
+                    let compiler_flags = format!("{} -target {}", bitcode_cflag, target);
+                    boringssl_cmake.define("CMAKE_ASM_FLAGS", &compiler_flags);
+                    // Work around hardcoded deployment target in cc crate by defining CMAKE_C_FLAGS
+                    // instead of using the cflag builder.
+                    boringssl_cmake.define("CMAKE_C_FLAGS", &compiler_flags);
+                    boringssl_cmake.define("CMAKE_CXX_FLAGS", &compiler_flags);
                 } else {
-                    ""
-                };
+                    // Normal iOS
 
-                let cflag = format!("{} {}", bitcode_cflag, target_cflag);
+                    // Hack for Xcode 10.1.
+                    let target_cflag = if arch == "x86_64" {
+                        "-target x86_64-apple-ios-simulator"
+                    } else {
+                        ""
+                    };
 
-                boringssl_cmake.define("CMAKE_ASM_FLAGS", &cflag);
-                boringssl_cmake.cflag(&cflag);
-            }
+                    let cflag = format!("{} {}", bitcode_cflag, target_cflag);
 
-            boringssl_cmake
-        }
+                    boringssl_cmake.define("CMAKE_ASM_FLAGS", &cflag);
+                    boringssl_cmake.cflag(&cflag);
+                }
+            }
 
-        _ => {
-            // Configure BoringSSL for building on 32-bit non-windows platforms.
-            if arch == "x86" && os != "windows" {
-                boringssl_cmake.define(
-                    "CMAKE_TOOLCHAIN_FILE",
-                    pwd.join(BORING_SSL_PATH)
-                        .join("src/util/32-bit-toolchain.cmake")
-                        .as_os_str(),
-                );
+            "windows" => {
+                if host.contains("windows") {
+                    // BoringSSL's CMakeLists.txt isn't set up for cross-compiling using Visual Studio.
+                    // Disable assembly support so that it at least builds.
+                    boringssl_cmake.define("OPENSSL_NO_ASM", "YES");
+                }
             }
 
-            boringssl_cmake
+            "linux" => match arch.as_str() {
+                "x86" => {
+                    boringssl_cmake.define(
+                        "CMAKE_TOOLCHAIN_FILE",
+                        pwd.join(BORING_SSL_PATH)
+                            .join("src/util/32-bit-toolchain.cmake")
+                            .as_os_str(),
+                    );
+                }
+                "aarch64" => {
+                    boringssl_cmake.define(
+                        "CMAKE_TOOLCHAIN_FILE",
+                        pwd.join("cmake/aarch64-linux.cmake").as_os_str(),
+                    );
+                }
+                _ => {
+                    eprintln!(
+                        "warning: no toolchain file configured by boring-sys for {}",
+                        target
+                    );
```

Comment:
if I understand correctly, this will print to stderr on cross comp to x86_64 which isn't an error right?